### PR TITLE
fix(twap): orders auth updating and rerending issues

### LINKED ISSUE #3530: Cancelled TWAP order is displayed on Open orders
1. Place TWAP order
2. Cancel the 1st part
3. Then cancel the parent order

**AR**: the TWAP order remains to be displayed on Open orders
Screenshot_1

**ER**: the order with 'cancelled' status should be moved to the History tab

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #3530 #3809

#3530 issue was introduced since the last multicall refactoring (#3416), the multicall in `useTwapOrdersAuthMulticall` returns a tuple of boolean, not just boolean.

#3809 issue also came from the multicall refactoring, the `useSWR` result shouldn't be used in hook dependencies, because it causes re-rendering issues.

I also found another rerendering issue in `TwapFormWidget`, there were two callbacks not wrapped in `useCallback`.

  # To Test

1. See #3530
2. For developers: add a console log into `TwapOrdersUpdater` body and check that there is no tons of calls

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/twap/hooks/useTwapOrdersAuthMulticall.ts [hunk 1]
```diff
@@ -29,15 +29,13 @@ export function useTwapOrdersAuthMulticall(
...
-    const loadedResults = results.data
-
     if (ordersInfo.length === 0) return EMPTY_AUTH_RESULT
 
-    if (results.isLoading || !loadedResults || loadedResults.length !== ordersInfo.length) return null
+    if (isLoading || !loadedResults || loadedResults.length !== ordersInfo.length) return null
 
     return ordersInfo.reduce((acc, val, index) => {
-      acc[val.id] = loadedResults[index]
+      acc[val.id] = loadedResults[index]?.[0]
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/twap/hooks/useTwapOrdersAuthMulticall.ts:37]: #3530 fix

File: apps/cowswap-frontend/src/modules/twap/hooks/useTwapOrdersAuthMulticall.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export function useTwapOrdersAuthMulticall(
     return ordersInfo.map(({ id }) => [safeAddress, id])
   }, [safeAddress, ordersInfo])
 
-  const results = useSingleContractMultipleData<boolean>(
+  const { data: loadedResults, isLoading } = useSingleContractMultipleData<[boolean]>(
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/twap/hooks/useTwapOrdersAuthMulticall.ts:23]: #3809 fix

Comment (alfetopito) [reply apps/cowswap-frontend/src/modules/twap/hooks/useTwapOrdersAuthMulticall.ts:23]: Why change the type from `boolean` to `[boolean]` ?
Was that also wrong?

**Review by elena-zh — APPROVED** [review]
Hey @shoom3301 , great job!
We still have some issues like #2883 or missing order cancellation modal (#3832), but they can be addressed as separate issues.

File: apps/cowswap-frontend/src/modules/twap/hooks/useTwapOrdersAuthMulticall.ts (hunk 2 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/twap/hooks/useTwapOrdersAuthMulticall.ts:23]: Yes, this is a part of https://github.com/cowprotocol/cowswap/pull/3831#discussion_r1486105396

**Review by alfetopito — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/twap/containers/TwapFormWidget/index.tsx [hunk 3]
```diff
@@ -102,6 +102,17 @@ export function TwapFormWidget() {
...
 
+  const onSlippageInput = useCallback(
+    (value: number | null) => updateSettingsState({ slippageValue: value }),
+    [updateSettingsState]
+  )
+  const onNumOfPartsInput = useCallback(
+    (value: number | null) => {
+      updateSettingsState({ numberOfPartsValue: value || DEFAULT_NUM_OF_PARTS })
+    },
+    [updateSettingsState]
```
Comment (anxolin) [inline apps/cowswap-frontend/src/modules/twap/containers/TwapFormWidget/index.tsx:113]: nit: not even sure if its a good idea, as your code is simpler, but here you make the two hooks to depend on the same `updateSettingsState`. Would this make the component to render twice if `updateSettingsState` (I'm not sure if React is smart enough in this case)

An alternative is to use `useMemo` who returns the 2 functions

Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/twap/containers/TwapFormWidget/index.tsx:113]: Good point, thanks!
https://github.com/cowprotocol/cowswap/pull/3847

**Pull request merged**